protocol/app/upgrades/v0.2.2/upgrades.go

```diff
@@ -0,0 +1,160 @@
+				clobPair.SubticksPerTick = 1e4
+				clobPair.QuantumConversionExponent = -7
+			case 15:
+				clobPair.SubticksPerTick = 1e8
+				clobPair.QuantumConversionExponent = -11
+			case 16:
+				clobPair.SubticksPerTick = 1e6
+				clobPair.QuantumConversionExponent = -9
+			case 17:
+				clobPair.SubticksPerTick = 1e3
+				clobPair.QuantumConversionExponent = -6
+			case 18:
+				clobPair.SubticksPerTick = 1e7
+				clobPair.QuantumConversionExponent = -10
+			case 19:
+				clobPair.SubticksPerTick = 1e5
+				clobPair.QuantumConversionExponent = -8
+			case 20:
+				clobPair.SubticksPerTick = 1e5
+				clobPair.QuantumConversionExponent = -8
+			case 21:
+				clobPair.SubticksPerTick = 1e6
+				clobPair.QuantumConversionExponent = -9
+			case 22:
+				clobPair.SubticksPerTick = 1e6
+				clobPair.QuantumConversionExponent = -9
+			case 23:
+				clobPair.SubticksPerTick = 1e6
+				clobPair.QuantumConversionExponent = -9
+			case 24:
+				clobPair.SubticksPerTick = 1e6
+				clobPair.QuantumConversionExponent = -9
+			case 25:
+				clobPair.SubticksPerTick = 1e7
+				clobPair.QuantumConversionExponent = -10
+			case 26:
+				clobPair.SubticksPerTick = 1e6
+				clobPair.QuantumConversionExponent = -9
+			case 27:
+				clobPair.SubticksPerTick = 1e6
+				clobPair.QuantumConversionExponent = -9
+			case 28:
+				clobPair.SubticksPerTick = 1e8
+				clobPair.QuantumConversionExponent = -11
+			case 29:
+				clobPair.SubticksPerTick = 1e7
+				clobPair.QuantumConversionExponent = -10
+			case 30:
+				clobPair.SubticksPerTick = 1e8
+				clobPair.QuantumConversionExponent = -11
+			case 31:
+				clobPair.SubticksPerTick = 1e7
+				clobPair.QuantumConversionExponent = -10
+			case 32:
+				clobPair.SubticksPerTick = 1e7
+				clobPair.QuantumConversionExponent = -10
+			}
+			clobKeeper.UnsafeSetClobPair(
+				ctx,
+				clobPair,
+			)
+		}
+
+		// Delete all stateful ordes.
+		removedOrderIds := make([]clobtypes.OrderId, 0)
+		placedStatefulOrders := clobKeeper.GetAllPlacedStatefulOrders(ctx)
+		for _, order := range placedStatefulOrders {
+			clobKeeper.MustRemoveStatefulOrder(ctx, order.OrderId)
+			removedOrderIds = append(removedOrderIds, order.OrderId)
+		}
+
+		untriggeredConditionalOrders := clobKeeper.GetAllUntriggeredConditionalOrders(ctx)
+		for _, order := range untriggeredConditionalOrders {
+			clobKeeper.MustRemoveStatefulOrder(ctx, order.OrderId)
+			removedOrderIds = append(removedOrderIds, order.OrderId)
+		}
+
+		// Purge invalid orders from memclob.
+		localValidatorOperationsQueue, _ := clobKeeper.MemClob.GetOperationsToReplay(ctx)
+		clobKeeper.MemClob.RemoveAndClearOperationsQueue(ctx, localValidatorOperationsQueue)
+
+		offchainUpdates := clobKeeper.MemClob.PurgeInvalidMemclobState(
+			ctx,
+			[]clobtypes.OrderId{},
+			[]clobtypes.OrderId{},
+			[]clobtypes.OrderId{},
+			removedOrderIds,
+			clobtypes.NewOffchainUpdates(),
+		)
+		clobKeeper.SendOffchainMessages(offchainUpdates, nil, metrics.SendPrepareCheckStateOffchainUpdates)
+
```

Comment:
@jayy04 @vincentwschau 

I think we also want to update the indexer. Can we try sending an Indexer event in the upgrade handler to see if this can trigger updates on the indexer side?

Reply:
Added logic to send UpdateClobPair events